terraform/modules/frontdoor/cloudfront.tf

```diff
@@ -44,6 +45,26 @@ resource "aws_cloudfront_distribution" "main" {
     }
   }
 
+  origin {
+    domain_name = aws_s3_bucket.maintenance_page_bucket.website_endpoint
+    origin_id   = local.maintenance_origin_id
+
+    custom_origin_config {
+      http_port              = 80
+      https_port             = 443
+      origin_protocol_policy = "http-only"
```

Comment:
When configured as an S3 website endpoint you can only use http

https://docs.aws.amazon.com/AmazonS3/latest/userguide/WebsiteEndpoints.html